cmd/checksum.go

```diff
@@ -0,0 +1,38 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/jpbetz/auger/pkg/data"
+	"github.com/spf13/cobra"
+)
+
+var checksumCmd = &cobra.Command{
+	Use:   "checksum",
+	Short: "Checksum a etcd keyspace.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return checksum()
+	},
+}
+
+type checksumOptions struct {
+	filename string
+	revision     int64
+}
+
+var checksumOpts *checksumOptions = &checksumOptions{}
+
+func init() {
+	RootCmd.AddCommand(checksumCmd)
+	checksumCmd.Flags().StringVarP(&checksumOpts.filename, "file", "f", "", "Bolt DB '.db' filename")
+	checksumCmd.Flags().Int64VarP(&checksumOpts.revision, "revision", "r", 0, "etcd revision to retrieve data at, if 0, the latest revision is used, defaults to 0")
+}
+
+func checksum() error {
+	hash, err := data.HashByRevision(checksumOpts.filename, checksumOpts.revision)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%d\n", hash)
```

Comment:
If we allow the use of a latest revision option, then we should print out which revision that hash is at.

Reply:
I've added revision to all output.